src/main/java/org/folio/rest/client/RestClient.java

```diff
@@ -0,0 +1,172 @@
+package org.folio.rest.client;
+
+import static javax.ws.rs.core.HttpHeaders.LOCATION;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+import static org.folio.config.Constants.OKAPI_URL;
+import static org.folio.util.RestUtils.ID;
+import static org.folio.util.RestUtils.SUCCESS_RESPONSE_PREDICATE;
+
+import java.util.Map;
+import java.util.concurrent.CompletionException;
+
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.common.WebClientFactory;
+
+import io.vertx.core.Context;
+import io.vertx.core.json.JsonObject;
+
+public class RestClient {
+   /**
+   * A common method to create a new entry in the storage based on the Json Object.
+   *
+   * @param recordData json to post
+   * @return future holding id of newly created entity Record or an exception if process failed
+   */
+  public <T> Future<T> post(T recordData, String endpoint, Class<T> responseType,
+                            RequestContext requestContext, Logger logger) {
+    var caseInsensitiveHeader = convertToCaseInsensitiveMap(requestContext.getHeaders());
+      if (logger.isDebugEnabled()) {
+        logger.debug("Trying to create object by endpoint '{}' and body '{}'", endpoint, JsonObject.mapFrom(recordData).encodePrettily());
+      }
+      return getVertxWebClient(requestContext.getContext())
+        .postAbs(buildAbsEndpoint(caseInsensitiveHeader, endpoint)).putHeaders(caseInsensitiveHeader)
+        .expect(SUCCESS_RESPONSE_PREDICATE)
+        .sendJson(recordData)
+        .map(bufferHttpResponse -> {
+          var id = verifyAndExtractRecordId(bufferHttpResponse);
+          return bufferHttpResponse.bodyAsJsonObject()
+            .put(ID, id)
+            .mapTo(responseType);
+        });
+  }
+
+  /**
+   * A common method to get an organization from the storage based on the Json Object.
+   *
+   * @return future jsonObject of created entity Record or an exception if failed
+   */
+  public Future<JsonObject> get(String endpoint, RequestContext requestContext, Logger logger) {
+    Promise<JsonObject> promise = Promise.promise();
```

Comment:
When you using promise - you should return promise.future()
See mod-orders RestClient for reference